programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -495,6 +498,128 @@ static void enroll(int argc, char *argv[]) {
     uv_run(l, UV_RUN_DEFAULT);
 }
 
+static tunnel_comand *cmd;
+
+static int dump_opts(int argc, char *argv[]) {
+    static struct option opts[] = {
+            {"identity", optional_argument, NULL, 'i'},
+            {"dump_path", optional_argument, NULL, 'p'},
+    };
+    int c, option_index, errors = 0;
+    optind = 0;
+
+    tunnel_ziti_dump *dump_options = calloc(1, sizeof(tunnel_ziti_dump));
+    cmd = calloc(1, sizeof(tunnel_comand));
+    cmd->command = TunnelCommand_ZitiDump;
+
+    while ((c = getopt_long(argc, argv, "i:p:",
+                            opts, &option_index)) != -1) {
+        switch (c) {
+            case 'i':
+                dump_options->id = optarg;
+                break;
+            case 'p':
+                dump_options->dump_path = realpath(optarg, NULL);
+                break;
+            default: {
+                fprintf(stderr, "Unknown option '%c'\n", c);
+                errors++;
+                break;
+            }
+        }
+    }
+    if (errors > 0) {
+        commandline_help(stderr);
+        exit(1);
+    }
+    size_t json_len;
+    cmd->data = tunnel_ziti_dump_to_json(dump_options, MODEL_JSON_COMPACT, &json_len);
+
+    return optind;
+}
+
+static void close_cmd_socket(uv_pipe_t* handle) {
+    uv_unref(handle);
+    uv_close(handle, NULL);
+}
+
+static void on_response(uv_stream_t *s, ssize_t len, const uv_buf_t *b) {
+    if (len > 0) {
+        printf("received response <%.*s>\n", (int) len, b->base);
+    } else {
+        printf("Read Response error %s\n", uv_err_name(len));
+        printf("received response <%.*s>\n", (int) len, b->base);
+    }
+    uv_read_stop(s);
+    close_cmd_socket(s);
+}
+
+void on_write(uv_write_t* req, int status) {
+    if (status < 0) {
+        printf("Could not sent message to the tunnel. Write error %s\n", uv_err_name(status));
+    } else {
+        puts("Message sent to the tunnel.");
+    }
+    free(req);
+}
+
+void send_message_to_pipe(uv_connect_t *connect, void *message, size_t datalen) {
+    printf("Message...%s\n", message);
```

Comment:
is this for debugging? should use ZITI_LOG then

Reply:
This message will be displayed in the command prompt, when you execute the ziti-edge-tunnel dump command from the commandline. So I used printf statements. I thought ZITI_LOG is used for writing to the log file.